tests/integration_tests/test_reproducibility.py

```diff
@@ -0,0 +1,182 @@
+import logging
+import os
+import random
+import tempfile
+
+import numpy as np
+import pytest
+import torch
+
+from ludwig.api import LudwigModel
+from ludwig.data.dataset_synthesizer import cli_synthesize_dataset
+
+INPUT_FEATURES = [
+    {"name": "num_1", "type": "number"},
+    {"name": "num_2", "type": "number"},
+]
+
+OUTPUT_FEATURES = [{"name": "y", "type": "number"}]
+
+CONFIG = {
+    "input_features": INPUT_FEATURES,
+    "output_features": OUTPUT_FEATURES,
+    "trainer": {"epochs": 2, "batch_size": 8},
+}
+
+
+@pytest.fixture(scope="module")
+def raw_dataset_fp() -> str:
+    """Generates dataset to be used in this test.
+
+    Returns (str):  file path string for dataset to use in this tests
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        raw_fp = os.path.join(tmpdir, "raw_data.csv")
+        random.seed(42)
+        cli_synthesize_dataset(64, INPUT_FEATURES + OUTPUT_FEATURES, raw_fp)
+        yield raw_fp
+
+
+@pytest.mark.parametrize("second_seed_offset", [0, 1])
+@pytest.mark.parametrize("random_seed", [1919, 31])
+def test_preprocess(raw_dataset_fp: str, random_seed: int, second_seed_offset: int) -> None:
+    """Test reproducibility of train/validation/test splits.
+
+    Args:
+        raw_dataset_fp (str): file path for data to be used as part of this test
+        random_seed(int): random seed integer to use for test
+        second_seed_offset(int): zero to use same random seed for second test, non-zero to use a different
+            seed for the second run.
+
+    Returns: None
+    """
+    # define Ludwig model
+    model1 = LudwigModel(config=CONFIG)
+
+    # preprocess the raw data set, specify seed
+    preprocessed_data1 = model1.preprocess(raw_dataset_fp, random_seed=random_seed)
+
+    # invoke torch random functions
+    torch.manual_seed(random_seed + 1)
+    torch.rand((5,))
```

Comment:
Is it necessary to set these seeds manually in the test for the test to pass?

Reply:
@w4nderlust wanted to confirm that an unrelated torch random operation, using an unrelated seed, did not affect Ludwig reproducibility.  That is the reason for this particular set of code.  Though I realize that with the introduction of the `second_seed_offset`, I need to account for this new construct.

Reply:
Thanks for the clarification.

If it is important to test that changing torch random seeds should not affect data preprocessing, then I would suggest putting this in it's own test:

```
@pytest.mark.parametrize("random_seed", [1919, 31])
def test_preprocess_ignores_torch_seed(raw_dataset_fp: str, random_seed: int) -> None:
    # define Ludwig model
    model1 = LudwigModel(config=CONFIG)

    # preprocess the raw data set, specify seed
    preprocessed_data1 = model1.preprocess(raw_dataset_fp, random_seed=random_seed)

    # Override torch seed.
    torch.manual_seed(random_seed + 1)
    torch.rand((5,))

    # define Ludwig model
    model2 = LudwigModel(config=CONFIG)
    
    # preprocess same raw data set with same seed
    preprocessed_data2 = model2.preprocess(raw_dataset_fp, random_seed=random_seed)

    # confirm data splits are reproducible
    for i in range(3):
        for k in preprocessed_data1[i].dataset:
            # same seeds should result in same output
            assert np.all(preprocessed_data1[i].dataset[k] == preprocessed_data2[i].dataset[k])
```

The decoupling makes the behavior that we are trying to test clearer (imo).